### PR TITLE
Address staticcheck warnings in prepared statement

### PR DESCRIPTION
**Pull request by saifalharthi**
Merged prepared statement code changes did not take into account
staticcheck warnings. This PR addresses those warnings.

Signed-off-by: Saif Alharthi <saif@saifalharthi.me>

### CODE REVIEW

File: go/mysql/query.go [hunk 1]
```diff
@@ -555,7 +555,7 @@ func (c *Conn) parseComStmtExecute(prepareData map[uint32]*PrepareData, data []b
 		}
 	}
 
-	newParamsBoundFlag, pos, ok := readByte(payload, pos)
+	newParamsBoundFlag, pos, _ := readByte(payload, pos)
```
Comment (deepthi) [inline go/mysql/query.go]: is it ok to ignore the ok here? what if it is false?
The rest look fine.

Comment (saifalharthi) [reply go/mysql/query.go]: I utilized the ok flag in the latest commit. Should be validating it correctly.

File: go/mysql/query.go [hunk 2]
```diff
@@ -556,7 +556,7 @@ func (c *Conn) parseComStmtExecute(prepareData map[uint32]*PrepareData, data []b
 	}
 
 	newParamsBoundFlag, pos, ok := readByte(payload, pos)
-	if newParamsBoundFlag == 0x01 {
+	if newParamsBoundFlag == 0x01 && ok {
```
Comment (deepthi) [inline go/mysql/query.go]: nit: should be `ok && ...`

Comment (saifalharthi) [reply go/mysql/query.go]: done :)

**Pull request merged**